### PR TITLE
Feat/filter notes

### PR DESCRIPTION
**Pull request by Debarshi95**
PR for UI fix, add notes filters functionality

### CODE REVIEW

File: src/components/common/Button/Button.css [hunk 1]
```diff
@@ -54,3 +54,10 @@
   background-color: var(--color-primary-disabled);
   cursor: initial;
 }
+
+@media screen and (max-width: 600px) {
```
Comment (tarunsankhla) [inline src/components/common/Button/Button.css:58]: Create const of media query pixels

File: src/components/dashboard/NoteCard/NoteCard.css [hunk 2]
```diff
@@ -2,17 +2,18 @@
...
   display: flex;
   flex-direction: column;
   position: relative;
   height: fit-content;
-  max-height: 32rem;
-  max-width: 21rem;
+  max-height: 36rem;
   overflow: hidden auto;
+  background-color: var(--color-white);
+  padding: 0.25rem;
```
Comment (tarunsankhla) [inline src/components/dashboard/NoteCard/NoteCard.css:13]: Avoid using decimal in rem

File: src/components/dashboard/NotePad/NotePad.css [hunk 3]
```diff
@@ -68,11 +70,17 @@
...
 }
 
-@media screen and (max-width: 768px) {
+@media screen and (max-width: 1024px) {
   .NotePad__root {
     max-width: 92vw;
   }
 }
+
+@media screen and (max-width: 600px) {
```
Comment (tarunsankhla) [inline src/components/dashboard/NotePad/NotePad.css:82]: Same here const for media query

File: src/components/dashboard/NotePad/NotePad.js [hunk 4]
```diff
@@ -70,7 +61,7 @@ const NotePad = () => {
     }
     const res = await handleAddNote({ content, cardColor, status, isPinned, tags });
     if (res) {
-      setDropdownObj({});
+      setDropdown(null);
```
Comment (tarunsankhla) [inline src/components/dashboard/NotePad/NotePad.js:64]: You can add res status check here

File: src/components/dashboard/NotePad/NotePad.css [hunk 5]
```diff
@@ -1,13 +1,15 @@
 .NotePad__root {
   width: 100%;
-  border: 1.5px solid var(--color-box-shadow);
   font-family: inherit;
   border-radius: 4px;
   position: relative;
   min-height: 10rem;
   border-radius: 4px;
```
Comment (tarunsankhla) [inline src/components/dashboard/NotePad/NotePad.css:7]: You can stack css based on order first display position width and then padding border

File: src/pages/App.js [hunk 6]
```diff
@@ -23,17 +22,6 @@ const App = () => {
               </PrivateRoute>
             }
           />
-          {['/labels', '/archive', '/trash'].map((route) => (
```
Comment (tarunsankhla) [inline src/pages/App.js:26]: This was good doing a map on array of private routes
Why did you remove

**Review by tarunsankhla — APPROVED** [review]
Overall it is great minor changes in css

File: src/components/auth/Input/Input.css [hunk 7]
```diff
@@ -1,6 +1,6 @@
 .Input__root {
   padding: 0.75rem 1rem;
```
Comment (manojsarna) [inline src/components/auth/Input/Input.css:2]: Suggestion : use px instead of rems if values computed to less than 10 px because you want a constant padding through screens for the particular component

File: src/components/auth/Input/Input.css [hunk 8]
```diff
@@ -1,6 +1,6 @@
...
-  outline: none;
+  outline: var(--color-main-bg);
   width: 100%;
   margin: 0.65rem 0;
```
Comment (manojsarna) [inline src/components/auth/Input/Input.css:5]: check for using px values here too

File: src/components/common/Button/Button.css [hunk 9]
```diff
@@ -27,7 +27,7 @@
   padding: 6px;
   border-radius: 50%;
   cursor: pointer;
-  min-width: 0;
+  min-width: 1em;
   margin-right: 4px;
```
Comment (manojsarna) [inline src/components/common/Button/Button.css:31]: Whats the significance of using em over here?

File: src/components/common/Searchbar/Searchbar.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+  priorityOptions,
+  statusOptions,
+} from 'constants/dropdownOptions';
+import './Searchbar.css';
+import { toCapitalize } from 'utils/helper-funcs';
+
+const Searchbar = () => {
+  const [select, setSelect] = useState(null);
+  const { tags = [] } = useTagContext();
+  const { dispatch } = useNoteFilterContext();
```
Comment (manojsarna) [inline src/components/common/Searchbar/Searchbar.js:22]: Suggestion : dispatch name is common, can use noteFilterDispatch to make it more readable throughout the code.

File: src/providers/NoteFilterProvider/NoteFilterProvider.js [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+  const getSortedNotes = (noteList = [], sortBy, search = '') => {
+    if (sortBy && sortBy === 'CREATED_AT') {
+      return noteList.filter((note) => note.createdAt);
+    }
+    if (sortBy) {
+      return noteList.filter((note) => note.priority === sortBy);
+    }
+    if (search) {
+      return noteList.filter((note) => sanitizeHtml(note.content)?.includes(search));
+    }
```
Comment (manojsarna) [inline src/providers/NoteFilterProvider/NoteFilterProvider.js:28]: what sanitize html doing here?

**Review by manojsarna — APPROVED** [review]
Overall the code is great!!

UI Review :

There are two delete buttons for a single note. Do Check!!

image

File: src/providers/NoteFilterProvider/NoteFilterProvider.js (hunk 11 above)
Comment (Debarshi95) [reply src/providers/NoteFilterProvider/NoteFilterProvider.js:28]: Need to search the content. Without sanitize it would add the html tags added by React Quill

File: src/components/common/Button/Button.css (hunk 9 above)
Comment (Debarshi95) [reply src/components/common/Button/Button.css:31]: It's width depends on its parent's font-size. That's why em here

**Comment by Debarshi95** [comment]
> Overall the code is great!!
> 
> UI Review :
> 
> There are two delete buttons for a single note. Do Check!!
> 
> image

The left icon is archive button. It would change the status from trash to archive. The right one changes the status to trash.

File: src/pages/App.js (hunk 6 above)
Comment (Debarshi95) [reply src/pages/App.js:26]: The paths and component is no longer required. This changes in this PR doesn't match the already production deploy
